gpflow/base.py

```diff
@@ -308,6 +306,24 @@ def _run_op(a, *args, **kwargs):
 tf.register_tensor_conversion_function(Parameter, lambda x, *args, **kwds: x.read_value())
 
 
+TensorType = Union[np.ndarray, tf.Tensor, tf.Variable, Parameter]
+"""
+Type alias for tensor-like types that are supported by most TensorFlow, NumPy and GPflow operations.
+
+NOTE: Union types like this do not work with the `register` method of multipledispatch's
+`Dispatcher` class. Instead use `TensorLikeTypes` for dispatching on tensor-like types.
+"""
+
+
+# We've left this as object until we've tested the performance consequences of using the full set
+# (np.ndarray, tf.Tensor, tf.Variable, Parameter), see https://github.com/GPflow/GPflow/issues/1434
+TensorLike: Final[Tuple[type, ...]] = (object,)
+"""
+:var TensorLike: Collection of tensor-like types for registering implementations with
```

Comment:
```suggestion
:const TensorLike: Collection of tensor-like types for registering implementations with
```
?

Reply:
nice. didn't know about that one

Reply:
oh, I think it's only for _referencing_ not declaring ...

for const
>     Reference a “defined” constant. This may be a Python variable that is not intended to be changed.

for var
> var, ivar, cvar: Description of a variable.

though it's not particularly clear

Reply:
i noticed cos pycharm doesn't highlight `const` like it highlights `var`